### DISCUSSION TITLE
Trying to understand remote repos

### DISCUSSION THREAD

**Discussion by DraxDomax** [community/community#22985] [New to GitHub]
<p>Hi, I have been using git as a local VCS for a little and now reading up on how to enable collaboration with the rest of the team. I think we’ll be using “remote repo”.<br>
Since I am new, I am not sure what is the team behaviour pattern here, maybe you could advice on that.</p>
<p>The section of the official git manual that got me confused is the following:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/016b0219aebe22694a860d56a669ed22e183faee.png" data-download-href="/uploads/short-url/cxJYi40BuUr8ntnahRvuXCSQMK.png?dl=1" title="remotegit.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/33070400/181074127-7f4e8642-706c-44d2-8917-946dde57b0bf.png" alt="remotegit.png" data-base62-sha1="cxJYi40BuUr8ntnahRvuXCSQMK" width="577" height="499" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/016b0219aebe22694a860d56a669ed22e183faee_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">remotegit.png</span><span class="informations">670×580 21.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<ol>
<li>
<p>Looking at their example of some ‘git remote’ output, I can see that there’s:<br>
URL/schacon/ticgit<br>
URL/paulboone/ticgit<br>
* What does that mean? That there are two different repos named samely (“ticgit”)?<br>
* Why would a team work like that?</p>
</li>
<li>
<p>I am only able to do ‘git remote’ within an existing repo, isn’t that a little diminishing the usefullness of ‘git remote add’, as (I think) it means that the remote aliases are only defined for each local repo?</p>
</li>
<li>
<p>In the example provided, they are defining an alias for “…/schacon/ticgit” but when I want to clone a repo that’s hosted in the cloud, the command I am told to copy-paste is more <em>like: </em>…/schacon/ticgit.git<br>
* Shouldn’t the manual tell us to refer to the full address (all the way including ticgit.git)?</p>
</li>
</ol>
<p>I thought there can be only one repo hosted online and all users have a clone locally that they modify and stage to commit to the one that’s in the cloud… Or I am totally missing the way git is used with a cloud distribution server?</p>
<p>Kudos and thanks!</p>

**Comment by pixeldesu** [comment, marked as answer]
<p>To answer all your questions one by one:</p>
<blockquote>
<p><a class="mention" href="/draxdomax">@draxdomax</a> wrote:</p>
</blockquote>
<blockquote>
<ol>
<li>Looking at their example of some ‘git remote’ output, I can see that there’s:<br>
URL/schacon/ticgit<br>
URL/paulboone/ticgit<br>
* What does that mean? That there are two different repos named samely (“ticgit”)?<br>
* Why would a team work like that?</li>
</ol>
</blockquote>
<p>The depicted set of repositories probably is displaying a fork-kind of nature here. So user “paulboone” might have forked off ticgit into his own repository to do a larger set of changes without having direct contribution rights to "schacon"s ticgit.</p>
<p>I’m not sure if teams would directly work like this, forking a core repository, but rather be used in the open source world and on GitHub moreso.</p>
<blockquote>
<p><a class="mention" href="/draxdomax">@draxdomax</a> wrote:</p>
</blockquote>
<blockquote>
<ol start="2">
<li>I am only able to do ‘git remote’ within an existing repo, isn’t that a little diminishing the usefullness of ‘git remote add’, as (I think) it means that the remote aliases are only defined for each local repo?</li>
</ol>
</blockquote>
<p>Yes, remotes only work inside git repositories, since a single remote always points to a single repository.</p>
<blockquote>
<p><a class="mention" href="/draxdomax">@draxdomax</a> wrote:</p>
</blockquote>
<blockquote>
<ol start="3">
<li>In the example provided, they are defining an alias for “…/schacon/ticgit” but when I want to clone a repo that’s hosted in the cloud, the command I am told to copy-paste is more <em>like: </em>…/schacon/ticgit.git<br>
* Shouldn’t the manual tell us to refer to the full address (all the way including ticgit.git)?</li>
</ol>
</blockquote>
<p>This depends on the server that is serving the git repositories, but usually the “.git” can be omitted! You can try this on GitHub, taking any repository path to clone and removing the “.git” part at the end, it still works!</p>
<blockquote>
<p><a class="mention" href="/draxdomax">@draxdomax</a> wrote:</p>
</blockquote>
<blockquote>
<p>I thought there can be only one repo hosted online and all users have a clone locally that they modify and stage to commit to the one that’s in the cloud… Or I am totally missing the way git is used with a cloud distribution server?</p>
</blockquote>
<p>There are many different ways to use git. Technically, you could, as you describe, put a single bare git repository on a server and have users clone that one and push their changes to it! In most cases though, people use systems like GitHub, GitLab or Bitbucket to host their repositories, and of course that allows more than a single repository to exist, but depending on project-scale or teams, one repository might be enough!</p>
<p>If you need advice for teams and appropriate workflows, I recommend you to check out these two workflows that have been widely adopted by teams all over the globe!:</p>
<ul>
<li><a href="https://nvie.com/posts/a-successful-git-branching-model/" rel="nofollow noopener">Git Flow</a></li>
<li><a href="https://guides.github.com/introduction/flow/" rel="nofollow noopener">GitHub Flow</a></li>
</ul>

**Comment by DraxDomax** [comment]
<p>How can I possibly thank you more than just saying that this was an amazing interaction for me to read your insight?</p>

**Comment by TG-009** [comment]
<p>Wow.</p>
<p>The Git Flow link is fantastic.  Very clear explaination of how a team should work.</p>
